src/librustdoc/formats/item_type.rs

```diff
@@ -48,6 +48,7 @@ crate enum ItemType {
     ProcAttribute = 23,
     ProcDerive = 24,
     TraitAlias = 25,
+    Generic = 26,
```

Comment:
Why are generic parameters treated as items?

Reply:
The pure generic ones, like `fn foo<T: Display + Debug>(a: T)`, in here, `a` cannot be both "Display" and "Debug", it needs to go through a level:

```
arg: [
  {
    name: "",
    generics: [
      {
        name: "Display"
        generics: []
      },
      {
        name: "Debug"
        generics: []
      }
    ]
  }
]
```

So in here, the argument with name `""` is a "pure" generic.

Reply:
But value parameters like `a: T` don't have item types (at least from looking at the `ItemType` enum definition), so how are they represented?

Reply:
Just with an empty name. If they are still present, might be interesting to remove them since they're useless for the search index.